app/javascript/runtimeBanner/RuntimeBanner.jsx

```diff
@@ -0,0 +1,145 @@
+/* global Runtime */
+
+import { h } from 'preact';
+
+const BANNER_DISMISS_KEY = 'runtimeBannerDismissed';
+
+/**
+ * A banner that will be displayed to provide a deep link into a specified
+ * ConsumerApp based on the Runtime context. If the banner is dismissed it will
+ * keep track of this in localStorage so it won't be rendered again.
+ */
+export const RuntimeBanner = () => {
+  const dismissBanner = () => {
+    localStorage.setItem(BANNER_DISMISS_KEY, true);
+    removeFromDOM();
+  };
+
+  const removeFromDOM = () => {
```

Comment:
Nitpick (non-blocking): Prefer function expressions over arrow function expressions in the componet, as there is no need to have access to the [enclosing lexical context's this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this#arrow_functions) since function components do no use `this`.